spec/unit/knife/bootstrap_options_spec.rb

```diff
@@ -0,0 +1,145 @@
+#
+# Author:: Kartik Null Cating-Subramanian(<ksubramanian@chef.io>)
+# Copyright:: Copyright (c) 2015 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'spec_helper'
+
+describe Chef::Knife::Bootstrap, :chef_gte_12_only do
+  before(:all) do
+    Chef::Config.reset
+  end
+
+  let(:bootstrap) { Chef::Knife::Bootstrap.new }
+  let(:win_bootstrap) { nil }
+  let(:opt_map) { {} }
+  let(:ref_ignore) { [] }
+  let(:win_ignore) { [] }
+
+  def compare_property(sym)
+    win_bootstrap.options.each do |win_key, win_val|
+      unless win_ignore.include?(win_key) || opt_map.include?(win_key) then
+        actual = win_val[sym]
+        expected = bootstrap.options[win_key][sym]
+        expect(actual).to eq(expected),
+          "#{win_key} flag's #{sym} property doesn't match
+
+expected: #{expected}
+     got: #{actual}"
+      end
+    end
+  end
+
+  # ref_opts, win_opts: Hashes of Mixlib::CLI options for core bootstrap and windows.
+  # opt_map: Hash of symbols in windows mapping to symbols in core.  Name checks are
+  #   ignored for these.
+  # ref_ignore: Options in core that we haven't implemented.
+  # win_ignore: Options in windows that aren't relevant to core.
+  shared_examples 'compare_options' do
+    it 'contains the option flags' do
+      opt_map.default_proc = proc { |map, key| key }
+      filtered_keys = (win_bootstrap.options.keys - win_ignore).map! { |key| opt_map[key] }
+
+      expect(filtered_keys).to match_array(bootstrap.options.keys - ref_ignore)
+    end
+
+    it 'uses the same long-name' do
+      compare_property(:long)
+    end
+
+    it 'uses the same short-name' do
+      compare_property(:short)
+    end
+
+    it 'uses the same description' do
+      compare_property(:description)
+    end
+
+    it 'uses the same default value' do
+      compare_property(:default)
+    end
+  end
+
+  context 'when compared to BootstrapWindowsWinrm' do
+    let(:win_bootstrap) { Chef::Knife::BootstrapWindowsWinrm.new }
+
+    let(:opt_map) { {
+      :msi_url => :bootstrap_url,
+      :encrypted_data_bag_secret => :secret,
+      :encrypted_data_bag_secret_file => :secret_file,
+      :winrm_user => :ssh_user,
+      :winrm_password => :ssh_password,
+      :winrm_port => :ssh_port,
+      :winrm_ssl_verify_mode => :host_key_verify,
+    }}
+    let(:ref_ignore) { [
```

Comment:
we should separate this into two arrays, one of things that we know are different and it's okay, and one that is a punch list of stuff we need to fix. and put a comment on them explaining that.